src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -134,14 +144,23 @@ func (r *ClusterReconciler) Reconcile(
 	if errors.IsNotFound(err) {
 		log.V(debugLevel).Info("Creating bootstrap StatefulSet")
 
-		if err = r.createBootstrapStatefulSet(ctx, &redpandaCluster, r.Scheme); err != nil {
+		if err = r.createBootstrapStatefulSet(ctx, &redpandaCluster, r.Scheme, redpandaCluster.Name+baseSuffix); err != nil {
 			log.Error(err, "Failed to create new bootstrap StatefulSet",
-				"Configmap.Namespace", redpandaCluster.Namespace, "StatefulSet.Name", redpandaCluster.Name+seedSuffix)
+				"Configmap.Namespace", redpandaCluster.Namespace, "StatefulSet.Name", redpandaCluster.Name)
 
 			return ctrl.Result{}, err
 		}
 	}
 
+	// Ensure StatefulSet #replicas equals cluster requirement.
+	if sts.Spec.Replicas != redpandaCluster.Spec.Replicas {
```

Comment:
Agree, go ahead. 
Since you are mentioning status, I noticed the list of nodes is also not being updated correctly (it has empty entries or old entries when decreasing replicas). If you work on the status you might notice.